R/utilities-output-schema.R

```diff
@@ -0,0 +1,64 @@
+#' Adds an interval to the output schema of the simulation
+#'
+#' @param simulation Simulation for which a new interval should be created
+#' @param startTime Start time of the interval in min
+#' @param endTime End time of the interval in min
+#' @param resolution Optional resolution in points/min (Default is 4 pts/hr)
+#' @param intervalName Optional Name of interval. If not specified, a unique name wil be assigned.
+#'
+#' @return Returns the interval created.
+#'
+#' @examples
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#'
+#' # Load the simulation
+#' sim <- loadSimulation(simPath, addToCache = FALSE, loadFromCache = FALSE)
+#'
+#' # clears the previous output schema
+#' clearIntervals(sim)
+#'
+#' # Adds a new interval starting at 1h and ending at 10h with a resolution of 10 points per hour
+#' addInterval(sim, 1 * 60, 10 * 60, 1 / 6)
+#'
+#' # Adds another interval starting at 10h and ending at 17h with a
+#' # default resolution and a specified name
+#' addInterval(sim, 10 * 60, 17 * 60, intervalName = "Second Interval")
+#' @export
+addInterval <- function(simulation, startTime, endTime, resolution = 4 / 60, intervalName = NULL) {

```

Comment:
First API method that I believe will be quite useful. 
I do not know if we need to define a value for the resolution by default

Reply:
I would say the resolution must always be suppplied by the caller.

Reply:
OK. I don't like the internal unit of pts/min
This is just a unit added for consistency in our internal system. 
Even min and max in [min] is weird.

Reply:
Why is it weird? I think it strongly depends on the scenarios simulated. In typical PBPK hours maybe the more common unit, but in QSP minutes make more sense I would say.